test/tools.cpp

```diff
@@ -30,10 +30,19 @@ TEST(tools, nbench_functional)
     const string model_path = file_util::path_join(SERIALIZED_ZOO, model);
 
     stringstream ss;
-
+#ifdef _WIN32
```

Comment:
Let's set up a thin compatibility layer file with these functions that are slightly different; for example `port_open` does the `popen` sort of thing, etc. And this is another area to have a separate `ifdef` for linux/macos instead of an `else`.

Reply:
Fixed.